modules/gapi/src/backends/ie/giebackend.cpp

```diff
@@ -130,11 +130,10 @@ inline IE::TensorDesc toIE(const cv::Mat &mat, cv::gapi::ie::TraitAs hint) {
         const size_t height   = mat.size().height;
         const size_t width    = mat.size().width;
 
-        const size_t strideH  = mat.step[0];
-
-        IE::BlockingDesc bdesc({1, height, width, channels} /* dims */,
-                               {0, 2, 3, 1} /* order for NHWC */,
-                               0 /* offset */,
+        const size_t strideH  = mat.step1();
+        IE::BlockingDesc bdesc({1, channels, height, width} /* dims */,
+                               {0, 2, 3, 1} /* order for NHWC   */,
+                               0            /* offset           */,
                                {0, 0, 0, 0} /* offsets for dims */,
                                {strideH * height, strideH, channels, 1} /* strides for dims */);
```

Comment:
> `{1, channels, height, width} /* dims */`
> ...
> `{strideH * height, strideH, channels, 1} /* strides for dims */`

Does `strides` correspond to `dims` after the change?

Reply:
Usually dims follow in logical order - `NCHW`, but strides describe actual layout.
The problem was in `step` we don't need to consider size of precision in bytes there.
For instance: CV_FLOAT32C3 - 3 elements in strides, but we used 3 * sizeof(float)

Reply:
@ilya-lavrenov Could you please take a look how to properly pass cv::Mat to `InferenceEngine` here? (or point to some example in IE / OMZ)

Reply:
https://docs.openvinotoolkit.org/latest/openvino_docs_IE_DG_Memory_primitives.html

Code after the fix looks good to me

Reply:
CC @ilyachur